aten/src/ATen/native/mkldnn/Pooling.cpp

```diff
@@ -226,10 +346,100 @@ Tensor& mkldnn_adaptive_avg_pool2d_out(
     Tensor& output,
     const Tensor& input,
     IntArrayRef output_size) {
-  AT_ERROR(
+  TORCH_CHECK(false,
       "mkldnn_adaptive_avg_pool2d_out: in-place mkldnn operations are not supported yet");
 }
 
+Tensor mkldnn_max_pool2d_backward(
+    const Tensor& grad_output,
+    const Tensor& output,
+    const Tensor& input,
+    IntArrayRef kernel_size,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    bool ceil_mode) {
+  return _mkldnn_pool2d_backward(
+      grad_output,
+      output,
+      input,
+      kernel_size,
+      stride,
+      padding,
+      dilation,
+      ceil_mode,
+      ideep::algorithm::pooling_max);
+}
+
+Tensor mkldnn_avg_pool2d_backward(
+    const Tensor& grad_output,
+    const Tensor& input,
+    IntArrayRef kernel_size,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    bool ceil_mode,
+    bool count_include_pad,
+    c10::optional<int64_t> divisor_override) {
+  return _mkldnn_pool2d_backward(
+      grad_output,
+      grad_output,
+      input,
+      kernel_size,
+      stride,
+      padding,
+      /*dilation*/ std::vector<int64_t>{1, 1},
+      ceil_mode,
+      count_include_pad ? ideep::algorithm::pooling_avg_include_padding
+                        : ideep::algorithm::pooling_avg_exclude_padding);
+}
+
+Tensor& mkldnn_avg_pool2d_backward_out(
+    Tensor & grad_input,
+    const Tensor & grad_output,
+    const Tensor & input,
+    IntArrayRef kernel_size,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    bool ceil_mode,
+    bool count_include_pad,
+    c10::optional<int64_t> divisor_override) {
+  TORCH_CHECK(false, "mkldnn_avg_pool2d_backward_out: in-place mkldnn operations are not supported yet");
+}
+
+Tensor mkldnn_adaptive_avg_pool2d_backward(
+    const Tensor& grad_output,
+    const Tensor& input) {
+  TORCH_CHECK(input.dim() == 4, "mkldnn_adaptive_avg_pool2d_backward: Expect 2D input");
```

Comment:
nit: maybe make the error message slightly more explicit that a 4D Tensor is expected.

Reply:
@albanD, the file use many **AT_ASSERTM**, any difference with **TORCH_CHECK**?  **AT_ASSERTM** will be deprecated, may can replace them.

Reply:
Yes you can replace all the AT_ASSERT with TORCH_CHECK